src/daemon/lldpd.c

```diff
@@ -201,6 +204,10 @@ lldpd_clone_port(struct lldpd_port *destination, struct lldpd_port *source)
 	memcpy(destination, cloned, sizeof(struct lldpd_port));
 	free(cloned);
 	free(output);
+
+	/* Initialize TX VLAN */
+	destination->p_vlan_tx_id = -1;
```

Comment:
I don't think you should initialize here. This would prevent applying a VLAN to all ports. The default port is a template for all future ports, to ensure they get a configuration applied to all ports.

Reply:
Could you give ma a hint where shall `p_vlan_tx_id` be initialized?

Reply:
The other place looks good (`lldpd_alloc_default_local_port()`)

Reply:
There is no need to have initialization any more. I introduced `p_vlan_tx_enabled`, which is `1` when the tag is valid.